agent-spawner: base64 encode the resalloc ticket DATA in env

praiskup opened the pull request:
Fixes: #144

siteshwar commented:
I have not tested these changes, but here are some general observations:

- Where is this variable named `RESALLOC_RESOURCE_DATA`?
- Where are `RESOURCE_NAME`, `RESOURCE_POOL_ID` etc. are set?
- Why `RESALLOC_RESOURCE_DATA` has to be base64 encoded? Is that because terminal output may contain binary characters?

praiskup commented:
These are set here:
https://github.com/praiskup/resalloc/blob/36a289d6e050c88dd5451dec9d838afa7779cbbb/resallocserver/manager.py#L40-L50

> Is that because terminal output may contain binary characters?

Yes.

siteshwar commented:
> These are set here:

Why they are not visible in the `env` output?

praiskup commented:
They are actually.  But resalloc != agent spawner.

siteshwar commented:
> They are actually. But resalloc != agent spawner.

Is [this comment](https://github.com/praiskup/resalloc/blob/36a289d6e050c88dd5451dec9d838afa7779cbbb/config/agent-spawner/config.yaml#L12-L14) valid for agent spawner? If not, it should be removed and we should mention `AGENT_SPAWNER_RESOURCE_DATA` instead.

praiskup commented:
Thank you for patiently punching me.  You are right; the comment was wrong.  I updated the docs, and fixed the fix.

Can you please take another look now?

siteshwar commented:
> Thank you for patiently punching me. You are right; the comment was wrong. I updated the docs, and fixed the fix.
> 
> Can you please take another look now?

LGTM. Although I have not tested these changes.

siteshwar commented on config/agent-spawner/config.yaml:
```diff
@@ -9,12 +9,14 @@
-#    # Prepare the agent.  Variable $RESALLOC_RESOURCE_DATA (base64 encoded)
-#    # is provided in the script environment.  Other variables like
-#    # RESOURCE_NAME, RESOURCE_POOL_ID, etc. are provided as well.
+#    # Prepare the agent.  Variable $AGENT_SPAWNER_RESOURCE_DATA (base64
```
The only thing I may add is that base64 decoded data may contain new lines. It could be tricky to debug, if you are not familiar with functioning of resalloc. It kept me confused for a while.

praiskup replied:
Indeed, the current code doesn't modify the stdout at all, and thus it contains the trailing newline.  We could implement some RFE for stdout post-processing if you think it is worth it.
